### ISSUE TITLE
Query is not showing anything for Region map

### ISSUE THREAD

**Issue by GabrielRondelli** [metabase/metabase#17880]
**Describe the bug**
Creating a question that uses counties to display a map, does not display the map.
This is the map: https://github.com/GabrielRondelli/geojson/blob/main/romania-counties.geojson that was used and was validated with geojson.io and https://mapshaper.org/.
The field values to identify the columns are: county in DB and Name_1 in the json map.
Tried an example using one value: Alba in county and Alba in the map file.

**Logs**
1. Metabase logs around that time:
```
[e867f568-f839-45d8-a8b8-f8f9ea450fdb] 2021-09-14T14:49:02+03:00 DEBUG metabase.server.middleware.log GET /api/session/properties 200 20.8 ms (2 DB calls) App DB connections: 3/15 Jetty threads: 7/50 (1 idle, 0 queued) (99 total active threads) Queries in flight: 0 (0 queued)
[e867f568-f839-45d8-a8b8-f8f9ea450fdb] 2021-09-14T14:49:02+03:00 DEBUG metabase.server.middleware.log GET /api/setup/admin_checklist 200 45.0 ms (11 DB calls) App DB connections: 0/15 Jetty threads: 5/50 (2 idle, 0 queued) (99 total active threads) Queries in flight: 0 (0 queued)
[e867f568-f839-45d8-a8b8-f8f9ea450fdb] 2021-09-14T14:49:45+03:00 DEBUG metabase.server.middleware.log GET /api/geojson 200 373.9 ms (0 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (4 idle, 0 queued) (105 total active threads) Queries in flight: 0 (0 queued)
[e867f568-f839-45d8-a8b8-f8f9ea450fdb] 2021-09-14T14:49:46+03:00 DEBUG metabase.server.middleware.log GET /api/geojson 200 37.7 ms (0 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (4 idle, 0 queued) (105 total active threads) Queries in flight: 0 (0 queued)
[e867f568-f839-45d8-a8b8-f8f9ea450fdb] 2021-09-14T14:49:47+03:00 DEBUG metabase.server.middleware.log GET /api/geojson 200 39.0 ms (0 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (4 idle, 0 queued) (105 total active threads) Queries in flight: 0 (0 queued)
[e867f568-f839-45d8-a8b8-f8f9ea450fdb] 2021-09-14T14:49:48+03:00 DEBUG metabase.server.middleware.log GET /api/geojson 200 35.7 ms (0 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (4 idle, 0 queued) (105 total active threads) Queries in flight: 0 (0 queued)
[e867f568-f839-45d8-a8b8-f8f9ea450fdb] 2021-09-14T14:49:48+03:00 DEBUG metabase.server.middleware.log GET /api/geojson 200 35.1 ms (0 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (4 idle, 0 queued) (105 total active threads) Queries in flight: 0 (0 queued)
[e867f568-f839-45d8-a8b8-f8f9ea450fdb] 2021-09-14T14:50:13+03:00 DEBUG metabase.server.middleware.log GET /api/geojson 200 165.8 ms (0 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (4 idle, 0 queued) (100 total active threads) Queries in flight: 0 (0 queued)
[e867f568-f839-45d8-a8b8-f8f9ea450fdb] 2021-09-14T14:50:15+03:00 DEBUG metabase.server.middleware.log PUT /api/setting/custom-geojson 204 16.2 ms (4 DB calls) App DB connections: 5/15 Jetty threads: 4/50 (4 idle, 0 queued) (100 total active threads) Queries in flight: 0 (0 queued)
[e867f568-f839-45d8-a8b8-f8f9ea450fdb] 2021-09-14T14:50:15+03:00 DEBUG metabase.server.middleware.log GET /api/setting 200 2.5 ms (0 DB calls) App DB connections: 1/15 Jetty threads: 4/50 (4 idle, 0 queued) (100 total active threads) Queries in flight: 0 (0 queued)
[e867f568-f839-45d8-a8b8-f8f9ea450fdb] 2021-09-14T14:50:15+03:00 DEBUG metabase.server.middleware.log GET /api/session/properties 200 14.5 ms (2 DB calls) App DB connections: 3/15 Jetty threads: 6/50 (2 idle, 0 queued) (100 total active threads) Queries in flight: 0 (0 queued)
[e867f568-f839-45d8-a8b8-f8f9ea450fdb] 2021-09-14T14:52:48+03:00 DEBUG metabase.server.middleware.log GET /api/util/bug_report_details 200 5.0 ms (1 DB calls) App DB connections: 4/15 Jetty threads: 3/50 (4 idle, 0 queued) (99 total active threads) Queries in flight: 0 (0 queued)
```
2. Browser console:
```
TypeError: Cannot read property 'display' of undefined
    at app-main.bundle.js?2d4f66d0e2c48cb54d7f:5
    at app-main.bundle.js?2d4f66d0e2c48cb54d7f:5
    at t.value (app-main.bundle.js?2d4f66d0e2c48cb54d7f:5)
    at t.value (app-main.bundle.js?2d4f66d0e2c48cb54d7f:5)
    at ss (vendor.bundle.js?2d4f66d0e2c48cb54d7f:498)
    at vendor.bundle.js?2d4f66d0e2c48cb54d7f:498
    at t.unstable_runWithPriority (vendor.bundle.js?2d4f66d0e2c48cb54d7f:506)
    at Wi (vendor.bundle.js?2d4f66d0e2c48cb54d7f:498)
    at Mu (vendor.bundle.js?2d4f66d0e2c48cb54d7f:498)
    at uu (vendor.bundle.js?2d4f66d0e2c48cb54d7f:498)
```
3.Browser: Google Chrome 92.0.4515.159 (Official Build) (64-bit) Windows 10

**To Reproduce**
1. Create a simple question
2. Group by counties
3. Check region map

**Expected behavior**
I want to see a heatmap of my clients based on counties

**Screenshots**
![image](https://user-images.githubusercontent.com/15094788/133254078-d4def490-38be-4a78-81b9-d7c665228e64.png)
![image](https://user-images.githubusercontent.com/15094788/133254122-2191f4d1-1263-42b2-bdab-0c330312d405.png)
![image](https://user-images.githubusercontent.com/15094788/133254959-b88c30a7-b3b3-4c9d-978b-4229e28fca5e.png)

**Severity**
Medium

**Metabase Diagnostic Info**

```json
{
  "browser-info": {
    "language": "en-US",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.159 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "Cp1252",
    "java.runtime.name": "Java(TM) SE Runtime Environment",
    "java.runtime.version": "1.8.0_301-b09",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "http://java.oracle.com/",
    "java.version": "1.8.0_301",
    "java.vm.name": "Java HotSpot(TM) Client VM",
    "java.vm.version": "25.301-b09",
    "os.name": "Windows 10",
    "os.version": "10.0",
    "user.language": "en",
    "user.timezone": "America/Los_Angeles"
  },
  "metabase-info": {
    "databases": [
      "mysql"
    ],
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "1.4.197 (2018-03-18)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "1.4.197 (2018-03-18)"
      }
    },
    "run-mode": "prod",
    "version": {
      "tag": "v0.40.2",
      "date": "2021-08-03",
      "branch": "release-x.40.x",
      "hash": "b884d29"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Comment by flamber** [comment]
Hi @GabrielRondelli 

I cannot reproduce on 0.40.4, but my first thought is #12962, so change the file.encoding=UTF-8 - I would also recommend [Java 11](https://www.metabase.com/docs/latest/operations-guide/java-versions.html), and [migrating away from H2](https://www.metabase.com/docs/latest/operations-guide/migrating-from-h2.html)

I would think that there probably should be more errors/warnings in the browser console.

`select 'Alba' as "County", 1 as "Count"`
![image](https://user-images.githubusercontent.com/1447303/133258357-f7307be3-8710-4d8d-821f-7ea4aae9bddb.png)

Faking data from Sample Dataset with Custom Column
`case([Category] = "Gizmo", "Alba", [Category] = "Widget", "Maramureș", [Category])`
![image](https://user-images.githubusercontent.com/1447303/133258148-1c0c3a08-c3e6-4edd-84b1-bb334455db80.png)
![image](https://user-images.githubusercontent.com/1447303/133258006-9691205f-3263-4bd8-a6af-5958ccad8912.png)

**Comment by GabrielRondelli** [comment]
Thank you! This has been solved by upgrading to 0.40.4, upgrading to Java 11 and changing the starting default encoding to UTF8 encoding.
